### PR TITLE
will now log progress in addition to just errors

### PR DESCRIPTION
**Pull request by irwaters**

### CODE REVIEW

File: lib/secrets.rb [hunk 1]
```diff
@@ -158,4 +161,8 @@ def read_vault_token(authfile_path)
   rescue OpenSSL::X509::CertificateError
     File.read(authfile_path)
   end
+
+  def log(msg)
+    puts "#{Time.now.to_i}: #{msg}" unless ENV["testing"]
```
Comment (grosser) [inline lib/secrets.rb]: to_i seems weird ... can we have a real date format ?
aren't our logs supposed to have dates automatically ... maybe we should add that instead ?

File: test/secrets_test.rb [hunk 2]
```diff
@@ -45,6 +45,7 @@ def response_body(body)
   end
 
   around do |test|
+    ENV["testing"] = "true"
```
Comment (grosser) [inline test/secrets_test.rb]: makes little sense in an around if it is not reset ... move to the base level ...

File: test/secrets_test.rb [hunk 3]
```diff
@@ -95,6 +96,13 @@ def response_body(body)
       File.read("SECRET").must_equal("foo")
     end
 
+    it 'logs' do
+      SecretsClient.any_instance.expects(:log).with('secrets found: SECRET,this/is/very/hidden')
```
Comment (grosser) [inline test/secrets_test.rb]: FYI capture can make this read nice ... but won't have a backtrace when it fails ... 
```Ruby
called = []
SecretsClient.any_instance.stubs(:log).capture(called)
called.must_equal ['xxxx']
```

File: lib/secrets.rb (hunk 1 above)
Comment (irwaters) [inline lib/secrets.rb]: can be whatever format people want.   not sure if the sidecar stuff get's logged anywhere else.

**Pull request merged**